core/src/main/scala/kafka/controller/ReplicaStateMachine.scala

```diff
@@ -297,6 +297,12 @@ class ReplicaStateMachine(controller: KafkaController) extends Logging {
   def partitionsAssignedToBroker(topics: Seq[String], brokerId: Int):Seq[TopicAndPartition] = {
     controllerContext.partitionReplicaAssignment.filter(_._2.contains(brokerId)).keySet.toSeq
   }
+
+  // Used only by test
+  def getReplicaState(partitionAndReplica: PartitionAndReplica): ReplicaState = {
```

Comment:
Could we just use the existing def replicasInState(topic: String, state: ReplicaState)?